lib/postgrex/stream.ex

```diff
@@ -58,14 +49,15 @@ defimpl Collectable, for: Postgrex.Stream do
 
   def into(%Stream{conn: %DBConnection{}} = stream) do
     %Stream{conn: conn, query: query, params: params, options: opts} = stream
+    opts = Keyword.put(opts, :copy, true)
```

Comment:
Instead of passing the stream, we will pass the query and the copy flag above. This means less protocols to implement and less work around streams.